plugins/sslservices/sslservices.cpp

```diff
@@ -203,51 +204,42 @@ class PKeyCache
 
         if (pkey)
         {
-            cache.emplace_front(hashc(reinterpret_cast<const byte *>(passphrase), passphraseLen, hashc(reinterpret_cast<const byte *>(key), keyLen, 0)), pkey);
-            if (cache.size() > OPENSSL_MAX_CACHE_SIZE)
-            {
-                EVP_PKEY_free(std::get<1>(cache.back()));
+            cache.emplace_front(hashc(reinterpret_cast<const byte *>(passphrase), passphraseLen, hashc(reinterpret_cast<const byte *>(key), keyLen, 0)), std::move(std::unique_ptr<EVP_PKEY, decltype(&EVP_PKEY_free)>(pkey, EVP_PKEY_free)));
+            if (cache.size() > SSLSERVICES_MAX_CACHE_SIZE)
                 cache.pop_back();
-            }
         }
         else
             failOpenSSLError("loading a pkey");
 
         return pkey;
```

Comment:
multi-threading: What happens if there are lots of calls to resolve items in the cache from other threads - so this item is evicted (and deleted) as soon as the function returns. May require a shared pointer.

Reply:
I don't think that is an issue since the caches are initialized with a thread_local storage specifier. Should the caches be shared among multiple threads?